frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/ArchiveExtractor.java

```diff
@@ -29,86 +34,101 @@
 }
 
 final class DefaultArchiveExtractor implements ArchiveExtractor {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultArchiveExtractor.class);
+
     private void prepDestination(File path, boolean directory) throws IOException {
         if (directory) {
             path.mkdirs();
         } else {
             if (!path.getParentFile().exists()) {
                 path.getParentFile().mkdirs();
             }
-            if(!path.getParentFile().canWrite()) {
-                throw new AccessDeniedException(
-                        String.format("Could not get write permissions for '%s'", path.getParentFile().getAbsolutePath()));
+            if (!path.getParentFile().canWrite()) {
+                throw new AccessDeniedException(String.format("Could not get write permissions for '%s'",
+                    path.getParentFile().getAbsolutePath()));
             }
         }
     }
 
     @Override
     public void extract(String archive, String destinationDirectory) throws ArchiveExtractionException {
         final File archiveFile = new File(archive);
-        FileInputStream fis = null;
-        try {
-            fis = new FileInputStream(archiveFile);
 
-            if("zip".equals(FileUtils.getExtension(archiveFile.getAbsolutePath()))){
+        try (FileInputStream fis = new FileInputStream(archiveFile)) {
+            if ("msi".equals(FileUtils.getExtension(archiveFile.getAbsolutePath()))) {
+                String command = "msiexec /a " + archiveFile.getAbsolutePath() + " /qn TARGETDIR=\""
```

Comment:
As far as I got to know, the msiexec command just extracts the archive to a given target folder. There is no software registry updated.